src/glados/db.py

```diff
@@ -0,0 +1,290 @@
+import logging
+from datetime import datetime
+from typing import List, Optional
+from uuid import uuid4
+
+from sqlalchemy import Column, DateTime, Integer, MetaData, String, Table, create_engine
+from sqlalchemy.dialects.postgresql import JSONB, UUID
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import Query, Session, sessionmaker
+
+Metadata = MetaData()
+Base = declarative_base(metadata=Metadata)
+
+TABLE_INTERACTIONS = "interactions"
+
+
+class DataStoreInteraction(Base):
+    __tablename__ = TABLE_INTERACTIONS
+    interaction_id = Column(UUID, primary_key=True, default=str(uuid4()))
+    ts = Column(DateTime, default=datetime.now())
+    bot = Column(String, nullable=False)
+    data = Column(JSONB, default=dict())
+    message_channel = Column(String, default=None)
+    message_ts = Column(DateTime, default=None)
+    ttl = Column(Integer, default=None)
+    followup_ts = Column(DateTime, default=None)
+    followup_action = Column(String, default=None)
+    cron_followup_action = Column(
+        String, default=None
+    )  # TODO(zpriddy): Do I need? Is this the same as followup_action?
+    followed_up = Column(DateTime, default=None)
+    followed_up_ts = Column(DateTime, default=None)
+
+    @property
+    def __sql_values__(self) -> dict:
+        values = self.__dict__.copy()
+        values.pop("_sa_instance_state")
+        values.pop("interaction_id")
+        return values
+
+    @property
+    def object(self):
+        return self
+
+    def update_row(self, session: Session):
+        session.query(DataStoreInteraction).filter(
+            DataStoreInteraction.interaction_id == self.interaction_id
+        ).update(self.__sql_values__)
+
+
+class DataStore:
+    def __init__(
+        self,
+        host: str,
+        username: str,
+        password: str,
+        port: int = 5432,
+        database: str = "glados",
+    ):
+        self.host = host
+        self.port = port
+        self.username = username
+        self.password = password
+        self.database = database
+
+        self.db = create_engine(
+            f"postgres://{self.username}:{self.password}@{self.host}:{self.port}/"
+            f"{self.database}"
+        )
+
+    def create_session(self) -> Session:
+        """Generate a new session with the existing connection.
+
+        Returns
+        -------
+        Session:
+            the session that was generated
+        """
+        session = sessionmaker(self.db)  # type: sessionmaker
```

Comment:
The `sessionmaker()` here is a factory. You could create it once in `__init__`.

`self.sessionmaker = sessionmaker(self.db)` in `__init__` and then just change this to `return self.sessionmaker()`